onnx/test/cpp/common_path_test.cc

```diff
@@ -8,63 +8,59 @@
 
 #include "onnx/common/path.h"
 
+#ifdef _WIN32
+// Only test clean_relative_path and normalize_separator on non-Windows
+// because Windows has its own implementation for them from std::filesystem::path.
+#else
 using namespace ONNX_NAMESPACE;
-
 namespace ONNX_NAMESPACE {
 namespace Test {
-namespace {
-std::string fix_sep(std::string path) {
-  std::string out = path;
-  normalize_separator(out);
-  return out;
-}
-} // namespace
 
 TEST(PathTest, CleanRelativePathTest) {
   // Already normal.
-  EXPECT_EQ(clean_relative_path("abc"), fix_sep("abc"));
-  EXPECT_EQ(clean_relative_path("abc/def"), fix_sep("abc/def"));
-  EXPECT_EQ(clean_relative_path("a/b/c"), fix_sep("a/b/c"));
-  EXPECT_EQ(clean_relative_path("."), fix_sep("."));
-  EXPECT_EQ(clean_relative_path(".."), fix_sep(".."));
-  EXPECT_EQ(clean_relative_path("../.."), fix_sep("../.."));
-  EXPECT_EQ(clean_relative_path("../../abc"), fix_sep("../../abc"));
+  EXPECT_EQ(clean_relative_path("abc"), "abc");
+  EXPECT_EQ(clean_relative_path("abc/def"), "abc/def");
+  EXPECT_EQ(clean_relative_path("a/b/c"), "a/b/c");
+  EXPECT_EQ(clean_relative_path("."), ".");
+  EXPECT_EQ(clean_relative_path(".."), "..");
+  EXPECT_EQ(clean_relative_path("../.."), "../..");
+  EXPECT_EQ(clean_relative_path("../../abc"), "../../abc");
   // Remove leading slash
-  EXPECT_EQ(clean_relative_path("/abc"), fix_sep("abc"));
-  EXPECT_EQ(clean_relative_path("/"), fix_sep("."));
+  EXPECT_EQ(clean_relative_path("/abc"), "abc");
```

Comment:
I don't understand this test case. It seems that we expect this function would force converting every abs path to relative path? But why "/abc" becomes "abc"? If the current working directory is "/", it makes sense. But, what if we are not in the root dir? 

And on Windows abs path **may** start with things like "C:" , not "/". Then in that case what this function `clean_relative_path` would do?

Reply:
Good question... I also found it is confusing when involving absolute path. Actually ONNX IR only allows using relative path to represent the location for external tensors: https://github.com/onnx/onnx/blob/main/docs/IR.md#external-tensor-data.

Therefore, I further add absolute path check in checker, add comment for clean_relative_path that it cannot work with absolute path, and remove absolute path tests in `[common_path_test.cc](https://github.com/onnx/onnx/pull/4470/files/903b9edb3b6b1a93abf3acebc4b1d1497980c3bf#diff-9a1899e6bc8ebf64e337ebc384ed395441ea42ba716a906906cb9a2150bb2732)` to prevent confusions.